src/components/Text/Annotated/AnnotatedText.tsx

```diff
@@ -2,50 +2,105 @@ import { BroccoliTextGeneric } from "../../../model/Broccoli.ts";
+ * - Line: piece of annotated text as received from broccoli, a broccoli 'line' can also contain a logical text
+ * - Marker: annotation of zero length, useful for marking locations of footnotes, page endings, etc.
+ * - Offset: start or end character index of an annotation in a line
+ *   - start index marks first character to include
+ *   - stop index marks first character to exclude
+ * - Segment: piece of line or annotation uninterrupted by annotation offsets
  */
 export const AnnotatedText = (props: TextHighlightingProps) => {
+  const { footnoteMarkerAnnotationTypes, pageMarkerAnnotationTypes } =
+    useProjectStore(projectConfigSelector);
   const annotations = useAnnotationStore().annotations;
 
   const { tier2, highlight } = useDetailUrlParams();
   const searchTerms = highlight;
+  // const isDummy = DUMMY_ID === tier2;
+  const isDummy = false;
+  const annotationTypesToHighlight =
+    useAnnotationStore().annotationTypesToHighlight;
+  const lines = props.text.lines;
 
-  const typesToHighlight = useAnnotationStore().annotationTypesToHighlight;
-  const positions = props.text.locations.annotations;
+  // TODO: clean up dummy data
+  const relativeAnnotations = isDummy
+    ? [
+        {
+          bodyId: annotations.find((a) => a.body.type === "tei:Ptr")!.body.id,
+          start: { line: 0, offset: 10 },
+          end: { line: 0, offset: -1 },
+        },
+      ]
+    : props.text.locations.annotations;
+  // No offsets when no relative annotations
+  const offsets: LineOffsets[] = [];
 
-  const annotationsToHighlight = getAnnotationsByTypes(
-    annotations,
-    typesToHighlight,
-  );
-  const lines = props.text.lines;
+  const singleLineAnnotations = annotations.filter(withTargetInSingleLine);
+
+  if (relativeAnnotations.length) {
+    const nestedAnnotations = singleLineAnnotations
+      .filter((a) => annotationTypesToHighlight.includes(a.body.type))
+      .map((annotation) =>
+        createNestedLineOffsets(annotation, relativeAnnotations, lines),
+      );
+    console.log("nestedAnnotations", nestedAnnotations);
+    offsets.push(...nestedAnnotations);
+  }
 
-  const offsets = annotationsToHighlight.map((a) =>
-    createLineOffsets(a, positions, lines),
-  );
   const searchRegex = createSearchRegex(searchTerms, tier2);
-  const searchOffsets = createLineSearchOffsets(lines, searchRegex);
-  offsets.push(...searchOffsets);
+  offsets.push(...createSearchOffsets(lines, searchRegex));
+  offsets.push(
+    ...singleLineAnnotations
+      .filter((a) => footnoteMarkerAnnotationTypes.includes(a.body.type))
+      .map((annotation) =>
+        createFootnoteMarkLineOffsets(annotation, relativeAnnotations),
+      ),
+  );
+  offsets.push(
+    ...annotations
+      .filter((a) => pageMarkerAnnotationTypes.includes(a.body.type))
+      .map((annotation) =>
+        createPageMarkLineOffsets(annotation, relativeAnnotations),
+      ),
+  );
+
+  console.log("AnnotatedText", {
```

Comment:
Console log

Reply:
Fixed in [8988b27](https://github.com/knaw-huc/textannoviz/pull/154/commits/8988b2734d0fa14a4d9f8206dc51dc3b236b484f)